packages/cli/src/pages/api/render.ts

```diff
@@ -2,19 +2,22 @@ import type { NextApiRequest, NextApiResponse } from "next";
 import { MjmlError } from "mjml-react";
 
 import renderTemplate from "../../util/renderTemplate";
+import { validateApiKey } from "src/util/validateApiKey";
 
 type Data = {
   error?: string; // api error messages
   html?: string;
   mjmlErrors?: MjmlError[];
 };
 
-export default function handler(
+export default async function handler(
   req: NextApiRequest,
   res: NextApiResponse<Data>
 ) {
   const { templateName, props } = "GET" === req.method ? req.query : req.body;
 
+  if (!(await validateApiKey(req, res))) return;
```

Comment:
can we put this check behind a config flag? I don't really think we should validate for render by default and don't want to break any existing users.

Maybe 

```
  if (process.env.MAILING_REQUIRE_API_KEY && !(await validateApiKey(req, res))) return;
```

Reply:
I was thinking the same thing, good suggestion